pkg/operation/botanist/secrets.go

```diff
@@ -600,6 +576,72 @@ func (b *Botanist) generateWantedSecrets(basicAuthAPIServer *secrets.BasicAuth,
 		},
 	})
 
+	kubernetesVersionGeq118, err := versionutils.CheckVersionMeetsConstraint(b.Shoot.KubernetesMajorMinorVersion, ">= 1.18")
+	if err != nil {
+		return nil, err
+	}
+
+	// If the shoot version is >= 1.18, the konnectivity apiserver proxy secrets are deployed, otherwise the VPN secrets are deployed
+	if kubernetesVersionGeq118 {
+		// Secret definitions  for konnectivity-server and konnectivity Agent
+		secretList = append(secretList,
+			&secrets.ControlPlaneSecretConfig{
+				CertificateSecretConfig: &secrets.CertificateSecretConfig{
+					Name: common.KonnectivityServerCertName,
+
+					CommonName:   common.KonnectivityServerCertName,
+					Organization: nil,
+					IPAddresses:  nil,
+					DNSNames:     konnectivityServerDNSNames,
+
+					CertType:  secrets.ServerCert,
+					SigningCA: certificateAuthorities[v1beta1constants.SecretNameCACluster],
+				},
+			},
+
+			// Secret definition for konnectivity-server
+			&secrets.ControlPlaneSecretConfig{
+				CertificateSecretConfig: &secrets.CertificateSecretConfig{
+					Name: common.KonnectivityAgentCertName,
+
+					CommonName:   common.KonnectivityAgentCertName,
+					Organization: nil,
+					IPAddresses:  []net.IP{},
+					DNSNames:     []string{},
```

Comment:
Any reason to initialize empty slices here?

Reply:
nope can set it to nil, or leave it out completely.